source/mesh_deformation/interface.cc

```diff
@@ -585,6 +719,123 @@ namespace aspect
+
+      const AffineConstraints<double> initial_deformation_constraints = make_initial_constraints();
+
+      QGauss<dim> quadrature(mesh_deformation_fe.degree + 1);
+      UpdateFlags update_flags = UpdateFlags(update_values | update_JxW_values | update_gradients);
+      FEValues<dim> fe_values (*sim.mapping, mesh_deformation_fe, quadrature, update_flags);
+
+      const unsigned int dofs_per_cell = fe_values.dofs_per_cell,
+                         dofs_per_face = sim.finite_element.dofs_per_face,
+                         n_q_points    = fe_values.n_quadrature_points;
+
+      std::vector<types::global_dof_index> cell_dof_indices (dofs_per_cell);
+      std::vector<unsigned int> face_dof_indices (dofs_per_face);
+      Vector<double> cell_vector (dofs_per_cell);
+      FullMatrix<double> cell_matrix (dofs_per_cell, dofs_per_cell);
+
+      // We are just solving a Laplacian in each spatial direction, so
+      // the degrees of freedom for different dimensions do not couple.
+      Table<2,DoFTools::Coupling> coupling (dim, dim);
+      coupling.fill(DoFTools::none);
+
+      for (unsigned int c=0; c<dim; ++c)
+        coupling[c][c] = DoFTools::always;
+
+      LinearAlgebra::SparseMatrix mesh_matrix;
+#ifdef ASPECT_USE_PETSC
+      LinearAlgebra::DynamicSparsityPattern sp(mesh_locally_relevant);
+#else
+      TrilinosWrappers::SparsityPattern sp (mesh_locally_owned,
+                                            mesh_locally_owned,
+                                            mesh_locally_relevant,
+                                            sim.mpi_communicator);
+#endif
+      DoFTools::make_sparsity_pattern (mesh_deformation_dof_handler,
+                                       coupling, sp,
+                                       initial_deformation_constraints, false,
+                                       Utilities::MPI::
+                                       this_mpi_process(sim.mpi_communicator));
+#ifdef ASPECT_USE_PETSC
+      SparsityTools::distribute_sparsity_pattern(sp,
+                                                 mesh_deformation_dof_handler.n_locally_owned_dofs_per_processor(),
+                                                 sim.mpi_communicator, mesh_locally_relevant);
+      sp.compress();
+      mesh_matrix.reinit (mesh_locally_owned, mesh_locally_owned, sp, sim.mpi_communicator);
+#else
+      sp.compress();
+      mesh_matrix.reinit (sp);
+#endif
+
+      // carry out the solution
+      FEValuesExtractors::Vector extract_vel(0);
+
+      LinearAlgebra::Vector rhs, deformation_solution;
+      rhs.reinit(mesh_locally_owned, sim.mpi_communicator);
+      deformation_solution.reinit(mesh_locally_owned, sim.mpi_communicator);
+
+      typename DoFHandler<dim>::active_cell_iterator cell = mesh_deformation_dof_handler.begin_active(),
+                                                     endc= mesh_deformation_dof_handler.end();
+      for (; cell!=endc; ++cell)
```

Comment:
``for (const auto& cell : mesh_deformation_dof_handler.active_cell_iterators())``